### ISSUE TITLE
Fix new join tests

### ISSUE THREAD

**Issue by mrks** [hyrise/hyrise#306]
Tim left us with a couple of join tests that fail:

```
RightJoinRefColumn
LeftJoinRefColumn
RightJoinEmptyRefColumn
LeftJoinEmptyRefColumn
```

I merged them with the current code base and they look to me as if they should work.

Also `JoinFullTest/2.JoinOnDictAndReferenceColumn, where TypeParam = opossum::JoinSortMerge` fails

**Comment by lawben** [comment]
The failing tests now run on all three Joins (see #503). I am currently working on `JoinOnDictAndReferenceColumn` for SMJ.

**Comment by lawben** [comment]
Fixed `JoinOnDictAndReferenceColumn` in #503. As soon as the PR is merged, this issue can be closed.

**Comment by mrks** [comment]
Thank you! You can also write `fixes #306` in the description of the PR so that the issue is automatically closed when the PR is merged.
